### DISCUSSION TITLE
Static container

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-rs#67] [General]
I just saw your `axum` example and I'm using `axum` too. I see you don't create a `Container` statically, but I wonder, for example, for having a more complex app state, like having a set of active websocket connections you must declare the `Container` statically. Am I correct?

**Comment by GregoryConrad** [comment]
Ideally the `Container` should never be `static`. You should try to put what you can behind capsules so that they naturally interact with each other and you just have to perform reads/writes from the container.

I haven't used web sockets in Rust, but I'd imagine there would be a nice way to expose them via capsules.

**Reply by busslina** [reply]
I willl study this issue in more deep.
